src/pages/admin/notices/edit/[id].vue

```diff
@@ -0,0 +1,150 @@
+      >
+        <div class="mb-3">
+          <AppInput
+            v-model="data.name"
+            :label="t('addNotice.form.name')"
+            required
+          />
+        </div>
+
+        <AppLabel :label="t('addNotice.form.startDateAndTime')" />
+        <div class="mb-3 flex">
+          <div>
+            <AppInput v-model="data.startDate" type="date" />
+          </div>
+          <div>
+            <AppInput v-model="data.startTime" type="time" />
+          </div>
+        </div>
+
+        <AppLabel :label="t('addNotice.form.expirationDateAndTime')" />
+        <div class="mb-3 flex">
+          <div>
+            <AppInput v-model="data.expirationDate" type="date" />
+          </div>
+          <div>
+            <AppInput v-model="data.expirationTime" type="time" />
+          </div>
+        </div>
+
+        <div class="mb-3">
+          <AppInput
+            v-model="data.text"
+            :label="t('addNotice.form.text')"
+            required
+          />
+        </div>
+
+        <div class="mb-3">
+          <AppInput
+            v-model="data.buttonText"
+            :label="t('addNotice.form.buttonText')"
+          />
+        </div>
+
+        <div class="mb-3">
+          <AppInput v-model="data.url" :label="t('addNotice.form.url')" />
+        </div>
+      </AppForm>
+    </div>
+  </template>
+</template>
+
+<script lang="ts" setup>
+import { format } from 'date-fns';
+import { ref, onBeforeMount, reactive } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useRouter } from 'vue-router';
+
+import AppForm from '../../../../components/forms/AppForm.vue';
+import AppInput from '../../../../components/forms/AppInput.vue';
+import AppLabel from '../../../../components/forms/AppLabel.vue';
+import PageTitle from '../../../../components/PageTitle.vue';
+import {
+  CreateNoticeData,
+  GetNoticeData,
+} from '../../../../utils/api/api.interface';
+import {
+  updateNotice,
+  fetchNotice,
+  createNotice,
+} from '../../../../utils/api/notice';
+
+const { t } = useI18n();
+const router = useRouter();
+
+const props = defineProps<{ id: string }>();
+const notice = ref<GetNoticeData | undefined>();
+
+const data = reactive({
+  name: '',
+  startDate: '',
+  startTime: '',
+  expirationDate: '',
+  expirationTime: '',
+  text: '',
+  buttonText: '',
+  url: '',
+});
+
+onBeforeMount(async () => {
+  notice.value = await fetchNotice(props.id as string);
+
+  data.name = notice.value.name;
+  data.startDate = notice.value.starts
+    ? format(notice.value.starts, 'yyyy-MM-dd')
+    : '';
+  data.startTime = notice.value.starts
+    ? format(notice.value.starts, 'HH:mm')
+    : '';
+  data.expirationDate = notice.value.expires
+    ? format(notice.value.expires, 'yyyy-MM-dd')
+    : '';
+  data.expirationTime = notice.value.expires
+    ? format(notice.value.expires, 'HH:mm')
+    : '';
+  data.text = notice.value.text;
+  data.buttonText = notice.value.buttonText ? notice.value.buttonText : '';
```

Comment:
What do you think of this vs using the `||` operator?
```ts
data.buttonText = notice.value.buttonText ? notice.value.buttonText : '';
data.buttonText = notice.value.buttonText || '';
```

I like the simplicity of the latter although understand that for people coming from other languages it can be confusing. It is however very common in JS/TS

Reply:
I'll use it, I was looking for a pattern like this :)